examples/routing_service_shapes_processor/cpp/README.rst

```diff
@@ -19,25 +19,32 @@ From the directory containing the example sources:
 
     mkdir build
     cd build
-    cmake .. -DCONNEXTDDS_DIR=$NDDSHOME\
+    cmake .. -DCONNEXTDDS_DIR=<Connext directory> \
              -DCONNEXTDDS_ARCH=<ARCH> \
              -DBUILD_SHARED_LIBS=ON
-             -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
 
 
 where:
 
-- the environment variable ``$NDDSHOME`` shall point to the installation
-  directory of RTI Connext.
+- ``<Connext directory>`` shall be replaced with the path to the installation
+  directory of *RTI Connext*.
 - ``<ARCH>`` shall be replaced with the target architecture where you are
   running the example (e.g., x64Darwin15clang7.0).
 
+.. note::
+
+    Certain CMake generators may require specific value for the target platform
+    through the ``-DCMAKE_GENERATOR_PLATFORM``. For example on Windows system,
+    you can specify the values ``x86`` or ``x64``.
+
 On successful execution of CMake, you will find different new files in your
-build directory, including a file with name ``Makefile``. This file represents
+build directory that will perform the build of the sources into a shared library.
+
+In Unix, you will find a file with name ``Makefile``. This file represents
 a Unix makefile that contains the rules to build a shared library containing
 the |PROCESSOR| plug-in implementation.
 
-Now run:
+To perform the build run:
```

Comment:
I think, it would be a better idea to suggest the ``cmake --build .`` command. It doesn't depend on the generator.

If you decide to suggest it, note you need to indicate the configuration for multi-configuration generators (like VS) with the ``--config`` parameter.

Reply:
Sorry, I don't know what is the ``-A`` parameter?

If you use ``cmake --build .`` you don't need to think about the command to build. CMake will execute the correct one for you.

Regarding the ``--config`` parameter for the multiconfiguration projects, it is needed. If you try to run ``cmake --build .`` without that parameter, CMake will use one by default. For instance, it will use ``Debug`` for Visual Studio Solutions. That will work when CMake was called using ``-DCMAKE_BUILD_TYPE=Debug`` but will fail if CMake was called using ``-DCMAKE_BUILD_TYPE=Release``

Reply:
You mentioned the "-A x64" in your previous comment,  when calling CMake.
I want to know the behavior of --config if it's not specified when running a multi-configuration generator. Will it fail or will it pick the default (usually VS allows a default configuration).